app/controllers/UserController.java

```diff
@@ -4,56 +4,82 @@
  * Created by Ejub on 31.1.2016.
  */
 
-
+import com.google.gson.Gson;
 import models.User;
 import play.libs.Json;
 import play.mvc.*;
-import utilities.Resources;
-import utilities.UserHelper;
+import utilities.*;
+import utilities.Error;
 import views.html.*;
-import com.fasterxml.jackson.databind.JsonNode;
 import play.mvc.BodyParser;
 
-
+/**
+ * Controller for User model.
+ */
 public class UserController extends Controller {
     /**
-     * Registers the user
+     * Registers the user after validating the input.
+     *
      * @return the authorization authToken in JSON format
      */
+    private PersistenceManager manager = new PersistenceManager();
+    private UserHelper userHelper = new UserHelper();
+
+
     @BodyParser.Of(BodyParser.Json.class)
     public Result register() {
-        UserHelper userHelper = new UserHelper(request().body().asJson());
+        User user = Json.fromJson(request().body().asJson(), User.class);
 
-        if(!userHelper.initializeUser()){
-            return badRequest(index.render(Resources.BAD_REQUEST_COULD_NOT_INITIALIZE));
+        if(userHelper.ifEmailExists(user)){
+            return badRequest(new Gson().toJson(new Error(Resources.BAD_REQUEST_EMAIL_EXISTS)));
         }
-        if(!userHelper.validateUser()){
-            return badRequest(index.render(Resources.BAD_REQUEST_INVALID_DATA));
+        if(!userHelper.initializeUser(user)){
+            return badRequest(new Gson().toJson(new Error(Resources.BAD_REQUEST_COULD_NOT_INITIALIZE)));
+        }
```

Comment:
initializeUser is a not self-explanatory method, as well as error message (what does it mean to consumer of the API that user cannot be initialized?)